tools/launch_release/build/Dockerfile

```diff
@@ -1,36 +1,52 @@
-FROM python:3.9-slim-bullseye
+FROM python:3.11-slim-bookworm
 LABEL maintainer='Weights & Biases <support@wandb.com>'
 
-# Install Go.
-COPY --from=golang:1.23-alpine /usr/local/go/ /usr/local/go/
-ENV PATH="/usr/local/go/bin:${PATH}"
-
-# Set Go environment paths
-ENV GOPATH /go
-ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
-
-# Set up Go workspace directories
-RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"
-
-# Install Rust using Rustup
-RUN apt-get update && apt-get install -y curl && \
-    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y && \
-    apt-get remove -y curl && apt-get -qy autoremove && \
-    apt-get clean && rm -r /var/lib/apt/lists/*
-ENV PATH="/root/.cargo/bin:${PATH}"
-
-
-# install git
+# Install dependencies
 RUN apt-get update && apt-get upgrade -y \
-    && apt-get install -y git gcc python3-dev \
+    && apt-get install -y git gcc python3-dev curl \
     && apt-get -qy autoremove \
     && apt-get clean && rm -r /var/lib/apt/lists/*
 
-# required pip packages
+# Download and install Go bootstrap
+# Go 1.23 ships with a package containing a critical security vulnerability.
+# We will use a bootstrap version of Go to build a patched version.
+COPY --from=golang:1.23-alpine /usr/local/go/ /usr/local/go-bootstrap
+ENV PATH="/usr/local/go/bin:${PATH}"
+
+# Set up Go environment for bootstrap
+ENV GOPATH=/go
+ENV PATH=$PATH:/usr/local/go-bootstrap/bin:$GOPATH/bin
+
+# Clone Go source and build with updated dependencies
+# remove bootstrap version of go
+RUN git clone https://go.googlesource.com/go /usr/local/go-source && \
+    cd /usr/local/go-source && \
+    git checkout go1.23.6 && \
+    # Update dependencies in the Go source
+    cd src && \
+    go mod edit -require=golang.org/x/crypto@v0.31.0 && \
+    go mod edit -require=golang.org/x/net@v0.36.0 && \
```

Comment:
Checking my understanding:

Specific versions of the `crypto` and `net` packages have security vulnerabilities, so we download go 1.23, update those packages to newer versions than the default, and then install go.

Additionally, the normal rust installation has security issues, so we add `--profile minimal --component cargo` to the installation script so we don't install the problematic packages.

Reply:
Re: go, yeah we use the bootstrap version of go to set up a go environment with updated packages, then uninstall the original version of go. You can't just update the packages in the original environment because the go build still refers to them and present "security vulnerabilities". This sucks, but should be removable when go stops shipping things referring to these old packages, or if Wiz improves the way it checks for security vulns.

yeah, and also we remove some root rust files that are not necessary in the image to remove some files that refer to rust packages with security vulnerabiltiies.

In reality, none of these present real security risks, but Wiz see's them as such and we need to resolve them anyways